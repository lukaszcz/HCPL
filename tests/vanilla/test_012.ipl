
let fib = \&x {
   let g = \&x \&a \&b {
      if == x 0 then a
      else g (- x 1) b (+ a b)
   };
   g x 0 1
};

let fibe = \&x {
   let g = \!x \&a \!b {
      if == x 0 then a
      else g (- x 1) b (+ a b)
   };
   g x 0 1
};

let fibee = \&x {
   let g = \!x \!a \!b {
      if == x 0 then a
      else g (- x 1) b (+ a b)
   };
   g x 0 1
};

print (fib 6);
print (fib 20);

print (fibe 6);
print (fibe 20);

print (fibee 6);
print (fibee 20);
