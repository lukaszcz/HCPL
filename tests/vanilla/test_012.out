8
6765
8
6765
8
6765
